src/configMaker.js

```diff
@@ -12,15 +12,56 @@ const makeConfig = (vars) => {
     }).join(".");
 
     const configObject = {};
-    _.set(configObject, assignPath, vars[configKey]);
+    set(configObject, assignPath, vars[configKey]);
 
     configObjectsArray.push(configObject);
     return true;
   });
 
-  return _.merge({}, ...configObjectsArray);
+  return merge({}, ...configObjectsArray);
+};
+
+/* Build configuration by merging defaults sections.
+ */
+const mergeConfigSections = (metadata, sections) => {
+    return Object.keys(sections).reduce(
+        (acc, key) => {
+            const section = {};
+            section[key] = sections[key](metadata);
+            return merge(acc, section);
+        },
+        metadata,
+    );
+};
+
+
+/* Build the full application configuration.
+ */
+const buildConfig = (defaults, vars) => {
+    const injector = getInjector();
+
+    // load environment variables
+    const environ = makeConfig(vars);
+
+    // generate metadata
+    const metadata = {
```

Comment:
I think this shouldn't be in the library, or at least, the defaults should be less specific

Reply:
I think the notion of a "core" graph and its metadata need to live somewhere. It's either here or in another, more fundamental library (e.g. `nodule-graph`). I could go either way, but it feels a bit early to split out another layer.

That said, if we keep this, we probably want a function for generating the metadata -- with this behavior as the default -- and a way to override.

Reply:
I'd also want to introduce the notion of "am I in unit tests?" and "am I running a local server?" here.